x-pack/platform/plugins/shared/streams_app/public/components/stream_list_view/tree_table.tsx

```diff
@@ -24,67 +49,217 @@ export function StreamsTreeTable({
   loading?: boolean;
 }) {
   const router = useStreamsAppRouter();
-  const items = React.useMemo(() => flattenTrees(asTrees(streams ?? [])), [streams]);
+  const { euiTheme } = useEuiTheme();
+
+  const {
+    dependencies: {
+      start: {
+        streams: { streamsRepositoryClient },
+      },
+    },
+  } = useKibana();
+  const { timeState } = useTimefilter();
+  const [docCounts, setDocCounts] = useState<Record<string, number>>({});
+
+  const streamNames = React.useMemo(() => (streams ?? []).map((s) => s.stream.name), [streams]);
+
+  useEffect(() => {
+    if (streamNames.length === 0) return;
+
+    const controller = new AbortController();
+    const fetchCounts = async () => {
+      const counts: Record<string, number> = {};
+      await Promise.all(
+        streamNames.map(async (name) => {
+          try {
+            const startStr = String(timeState.start);
+            const endStr = String(timeState.end);
+
+            const { details } = await streamsRepositoryClient.fetch(
+              'GET /internal/streams/{name}/_details',
+              {
+                params: {
+                  path: { name },
+                  query: { start: startStr, end: endStr },
+                },
+                signal: controller.signal,
+              }
+            );
+            counts[name] = details?.count ?? 0;
+          } catch (e) {
+            if (!controller.signal.aborted) {
+              counts[name] = 0;
+            }
+          }
+        })
+      );
+      if (!controller.signal.aborted) {
+        setDocCounts(counts);
+      }
+    };
+    fetchCounts();
+    return () => {
+      controller.abort();
+    };
+  }, [streamNames, timeState, streamsRepositoryClient]);
```

Comment:
I think this is getting a bit too messy. We're now fetching the list of streams and then making two API calls for each stream in the list. One for the doc count and another one for the histogram. I think it's cleaner and more performant to update the BFF (backend for frontend) endpoint that we already have with document counts instead of fetching these individually on the client: `GET /internal/streams`

Reply:
I removed the logic completely from `tree_table`

Reply:
I removed the logic, so we’re no longer sorting the Documents column. Do you still think it’s worth updating the endpoint?

Reply:
No, let's do that when we implement document count sorting